runtime/minor_gc.c

```diff
@@ -518,13 +519,97 @@ void caml_empty_minor_heap_domain_clear (struct domain* domain, void* unused)
   clear_table ((struct generic_table *)&minor_tables->ephe_ref);
   clear_table ((struct generic_table *)&minor_tables->custom);
 
+  return;
+}
+
+void caml_adjust_global_minor_heap(int participating_domains) {
+  // Calculate the size of the existing mapping
+  int global_minor_heap_size = caml_global_minor_heap_limit - caml_global_minor_heap_start;
+  // Now using the number of participating domains, we calculate the new size
+  int new_global_minor_heap_size = participating_domains*Bsize_wsize(global_minor_heap_wsz_per_domain) + caml_global_minor_heap_start;
+
+  if( global_minor_heap_size != new_global_minor_heap_size ) {
+    caml_mem_decommit((char*)caml_global_minor_heap_start, global_minor_heap_size);
+    caml_mem_commit((char*)caml_global_minor_heap_start, new_global_minor_heap_size);
+
+    caml_global_minor_heap_limit = caml_global_minor_heap_start + new_global_minor_heap_size;
+  }
+
+  atomic_store_explicit(&caml_global_minor_heap_ptr,
+            caml_global_minor_heap_start,
+            memory_order_release);
+}
+
+int caml_replenish_minor_heap()
+{
+  caml_domain_state* domain_state = Caml_state;
+  uintnat cached_global_minor_heap_ptr;
+  uintnat new_alloc_ptr;
+
+  uintnat minor_buffer_wsize = global_minor_heap_wsz_per_domain >> 3;
```

Comment:
I had a concern here that a user might somehow set the minor heap size small and then we would have a minor heap that couldn't hold even the smallest allocation. Do we need a guard here or in the initialization of `global_minor_heap_wsz_per_domain`?